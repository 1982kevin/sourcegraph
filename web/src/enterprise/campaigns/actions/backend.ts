import { map } from 'rxjs/operators'
import { dataOrThrowErrors, gql } from '../../../../../shared/src/graphql/graphql'
import { queryGraphQL, mutateGraphQL } from '../../../backend/graphql'
import { Observable } from 'rxjs'
import {
    ID,
    IEmptyResponse,
    IActionExecution,
    IActionsOnQueryArguments,
    IActionConnection,
    IAction,
    ActionJobState,
    IActionJob,
    ICreateActionOnMutationArguments,
} from '../../../../../shared/src/graphql/schema'
import { PreviewFileDiffFields, FileDiffHunkRangeFields, DiffStatFields } from '../../../backend/diff'

export async function retryActionJob(actionJobID: ID): Promise<IEmptyResponse | null> {
    const result = await mutateGraphQL(
        gql`
            mutation RetryActionJob($actionJob: ID!) {
                retryActionJob(actionJob: $actionJob) {
                    alwaysNil
                }
            }
        `,
        { actionJob: actionJobID }
    ).toPromise()
    return dataOrThrowErrors(result).retryActionJob
}

export const fetchActionExecutionByID = (actionExecution: ID): Observable<IActionExecution | null> =>
    queryGraphQL(
        gql`
            query ActionExecutionByID($actionExecution: ID!) {
                node(id: $actionExecution) {
                    __typename
                    ... on ActionExecution {
                        id
                        action {
                            id
                            name
                            schedule
                            cancelPreviousScheduledExecution
                            savedSearch {
                                id
                                description
                            }
                            campaign {
                                id
                                name
                            }
                        }
                        definition {
                            steps
                            actionWorkspace {
                                name
                            }
                            env {
                                key
                            }
                        }
                        invokationReason
                        status {
                            errors
                            state
                            pendingCount
                            completedCount
                        }
                        executionStart
                        executionEnd
                        patchSet {
                            id
                        }
                        jobs {
                            totalCount
                            nodes {
                                id
                                repository {
                                    name
                                }
                                baseRevision
                                state
                                runner {
                                    id
                                    name
                                    description
                                    state
                                }
                                executionStart
                                executionEnd
                                log
                                diff {
                                    fileDiffs {
                                        nodes {
                                            ...PreviewFileDiffFields
                                        }
                                        totalCount
                                        pageInfo {
                                            hasNextPage
                                        }
                                        diffStat {
                                            ...DiffStatFields
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }

            ${PreviewFileDiffFields}

            ${FileDiffHunkRangeFields}

            ${DiffStatFields}
        `,
        { actionExecution }
    ).pipe(
        map(dataOrThrowErrors),
        map(({ node }) => {
            if (!node) {
                return null
            }
            if (node.__typename !== 'ActionExecution') {
                throw new Error(`The given ID is a ${node.__typename}, not a ActionExecution`)
            }
            return node
        })
    )

export const queryActions = ({ first }: IActionsOnQueryArguments): Observable<IActionConnection> =>
    queryGraphQL(
        gql`
            query Actions($first: Int) {
                actions(first: $first) {
                    totalCount
                    nodes {
                        id
                        name
                        savedSearch {
                            description
                        }
                        schedule
                        actionExecutions {
                            totalCount
                            nodes {
                                id
                                invokationReason
                                status {
                                    errors
                                    state
                                    pendingCount
                                    completedCount
                                }
                                patchSet {
                                    id
                                }
                            }
                        }
                    }
                }
            }
        `,
        { first }
    ).pipe(
        map(dataOrThrowErrors),
        map(data => data.actions)
    )

export const fetchActionByID = (action: ID): Observable<IAction | null> =>
    queryGraphQL(
        gql`
            query ActionByID($action: ID!) {
                node(id: $action) {
                    __typename
                    ... on Action {
                        id
                        name
                        savedSearch {
                            description
                        }
                        definition {
                            steps
                            actionWorkspace {
                                name
                            }
                            env {
                                key
                                value
                            }
                        }
                        schedule
                        actionExecutions {
                            totalCount
                            nodes {
                                id
                                invokationReason
                                status {
                                    errors
                                    state
                                    pendingCount
                                    completedCount
                                }
                                patchSet {
                                    id
                                }
                            }
                        }
                    }
                }
            }
        `,
        { action }
    ).pipe(
        map(dataOrThrowErrors),
        map(({ node }) => {
            if (!node) {
                return null
            }
            if (node.__typename !== 'Action') {
                throw new Error(`The given ID is a ${node.__typename}, not a Action`)
            }
            return node
        })
    )

export async function createActionExecution(actionID: ID): Promise<IActionExecution> {
    const result = await mutateGraphQL(
        gql`
            mutation CreateActionExecution($action: ID!) {
                createActionExecution(action: $action) {
                    id
                }
            }
        `,
        { action: actionID }
    ).toPromise()
    return dataOrThrowErrors(result).createActionExecution
}

export async function createAction(args: ICreateActionOnMutationArguments): Promise<IAction> {
    const result = await mutateGraphQL(
        gql`
            mutation CreateAction($name: String!, $definition: JSONCString!) {
                createAction(name: $name, definition: $definition) {
                    id
                }
            }
        `,
        args
    ).toPromise()
    return dataOrThrowErrors(result).createAction
}

export async function updateAction(action: ID, newDefinition: string): Promise<IAction> {
    const result = await mutateGraphQL(
        gql`
            mutation UpdateAction($action: ID!, $newDefinition: JSONCString!) {
                updateAction(action: $action, newDefinition: $newDefinition) {
                    id
                }
            }
        `,
        { newDefinition, action }
    ).toPromise()
    return dataOrThrowErrors(result).updateAction
}

export async function updateActionJob(actionJob: ID, { state }: { state?: ActionJobState }): Promise<IActionJob> {
    const result = await mutateGraphQL(
        gql`
            mutation UpdateActionJob($actionJob: ID!, $state: ActionJobState) {
                updateActionJob(actionJob: $actionJob, state: $state) {
                    id
                }
            }
        `,
        { state, actionJob }
    ).toPromise()
    return dataOrThrowErrors(result).updateActionJob
}

export async function cancelActionExecution(actionExecution: ID): Promise<IEmptyResponse | null> {
    const result = await mutateGraphQL(
        gql`
            mutation CancelActionExecution($actionExecution: ID!) {
                cancelActionExecution(actionExecution: $actionExecution) {
                    alwaysNil
                }
            }
        `,
        { actionExecution }
    ).toPromise()
    return dataOrThrowErrors(result).cancelActionExecution
}
