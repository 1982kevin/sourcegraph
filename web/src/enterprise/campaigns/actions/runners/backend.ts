import * as GQL from '../../../../../../shared/src/graphql/schema'
import { queryGraphQL } from '../../../../backend/graphql'
import { gql, dataOrThrowErrors } from '../../../../../../shared/src/graphql/graphql'
import { map } from 'rxjs/operators'
import { Observable } from 'rxjs'

export function queryActionJobs(
    args: GQL.IActionJobsOnQueryArguments
): Observable<Pick<GQL.IActionJobConnection, 'totalCount'>> {
    return queryGraphQL(
        gql`
            query ActionJobs($state: ActionJobState!, $first: Int) {
                actionJobs(state: $state, first: $first) {
                    totalCount
                }
            }
        `,
        args
    ).pipe(
        map(dataOrThrowErrors),
        map(data => data.actionJobs)
    )
}
