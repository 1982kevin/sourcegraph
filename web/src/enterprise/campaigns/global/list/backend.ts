import { map } from 'rxjs/operators'
import { dataOrThrowErrors, gql } from '../../../../../../shared/src/graphql/graphql'
import { queryGraphQL } from '../../../../backend/graphql'
import { Observable } from 'rxjs'
import { Connection } from '../../../../components/FilteredConnection'
import { CampaignNodeProps } from '../../list/CampaignNode'
import { CampaignsResult, CampaignsVariables } from '../../../../graphql-operations'

export const queryCampaigns = ({
    first,
    state,
    viewerCanAdminister,
}: CampaignsVariables): Observable<Connection<CampaignNodeProps['node']>> =>
    queryGraphQL<CampaignsResult>(
        gql`
            query Campaigns($first: Int, $state: CampaignState, $viewerCanAdminister: Boolean) {
                campaigns(first: $first, state: $state, viewerCanAdminister: $viewerCanAdminister) {
                    nodes {
                        id
                        name
                        description
                        createdAt
                        closedAt
                        author {
                            username
                        }
                        changesets {
                            stats {
                                open
                                closed
                                merged
                            }
                        }
                    }
                    totalCount
                }
            }
        `,
        { first, state, viewerCanAdminister }
    ).pipe(
        map(dataOrThrowErrors),
        map(data => data.campaigns)
    )
