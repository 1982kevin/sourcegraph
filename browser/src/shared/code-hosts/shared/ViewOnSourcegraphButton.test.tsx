import { ViewOnSourcegraphButton } from './ViewOnSourcegraphButton'
import { HTTPStatusError } from '../../../../../shared/src/backend/fetch'
import * as React from 'react'
import { noop } from 'lodash'
import { mount } from 'enzyme'

describe('<ViewOnSourcegraphButton />', () => {
    describe('repository exists on the instance', () => {
        it('renders a link to the repository on the Sourcegraph instance', () => {
            expect(
                mount(
                    <ViewOnSourcegraphButton
                        codeHostType="test-codehost"
                        sourcegraphURL="https://test.com"
                        getContext={() => ({ rawRepoName: 'test', privateRepository: false })}
                        className="test"
                        repoExistsOrError={true}
                        minimalUI={false}
                    />
                ).children()
            ).toMatchSnapshot()
        })

        it('renders nothing in minimal UI mode', () => {
            expect(
                mount(
                    <ViewOnSourcegraphButton
                        codeHostType="test-codehost"
                        sourcegraphURL="https://test.com"
                        getContext={() => ({ rawRepoName: 'test', privateRepository: false })}
                        className="test"
                        repoExistsOrError={true}
                        minimalUI={true}
                    />
                ).children()
            ).toMatchSnapshot()
        })

        it('renders a link with the revision when provided', () => {
            expect(
                mount(
                    <ViewOnSourcegraphButton
                        codeHostType="test-codehost"
                        sourcegraphURL="https://test.com"
                        getContext={() => ({
                            rawRepoName: 'test',
                            revision: 'test',
                            privateRepository: false,
                        })}
                        className="test"
                        repoExistsOrError={true}
                        minimalUI={false}
                    />
                ).children()
            ).toMatchSnapshot()
        })
    })

    describe('repository does not exist on the instance', () => {
        it('renders "Configure Sourcegraph" button when pointing at sourcegraph.com', () => {
            expect(
                mount(
                    <ViewOnSourcegraphButton
                        codeHostType="test-codehost"
                        sourcegraphURL="https://sourcegraph.com"
                        getContext={() => ({
                            rawRepoName: 'test',
                            revision: 'test',
                            privateRepository: false,
                        })}
                        className="test"
                        repoExistsOrError={false}
                        onConfigureSourcegraphClick={noop}
                        minimalUI={false}
                    />
                ).children()
            ).toMatchSnapshot()
        })

        it('renders a "Repository not found" button when not pointing at sourcegraph.com', () => {
            expect(
                mount(
                    <ViewOnSourcegraphButton
                        codeHostType="test-codehost"
                        sourcegraphURL="https://sourcegraph.test"
                        getContext={() => ({
                            rawRepoName: 'test',
                            revision: 'test',
                            privateRepository: false,
                        })}
                        className="test"
                        repoExistsOrError={false}
                        onConfigureSourcegraphClick={noop}
                        minimalUI={false}
                    />
                ).children()
            ).toMatchSnapshot()
        })
    })

    describe('existence could not be determined ', () => {
        describe('because of an authentication failure', () => {
            for (const minimalUI of [true, false]) {
                describe(`minimalUI = ${String(minimalUI)}`, () => {
                    it('renders a sign in button if showSignInButton = true', () => {
                        expect(
                            mount(
                                <ViewOnSourcegraphButton
                                    codeHostType="test-codehost"
                                    sourcegraphURL="https://test.com"
                                    getContext={() => ({
                                        rawRepoName: 'test',
                                        revision: 'test',
                                        privateRepository: false,
                                    })}
                                    showSignInButton={true}
                                    className="test"
                                    repoExistsOrError={new HTTPStatusError(new Response('', { status: 401 }))}
                                    minimalUI={minimalUI}
                                />
                            ).children()
                        ).toMatchSnapshot()
                    })
                })
            }
        })

        describe('because of an unknown error', () => {
            it('renders a button with an error label', () => {
                expect(
                    mount(
                        <ViewOnSourcegraphButton
                            codeHostType="test-codehost"
                            sourcegraphURL="https://test.com"
                            getContext={() => ({
                                rawRepoName: 'test',
                                revision: 'test',
                                privateRepository: false,
                            })}
                            showSignInButton={true}
                            className="test"
                            repoExistsOrError={new Error('Something unknown happened!')}
                            minimalUI={false}
                        />
                    ).children()
                ).toMatchSnapshot()
            })
        })
    })
})
