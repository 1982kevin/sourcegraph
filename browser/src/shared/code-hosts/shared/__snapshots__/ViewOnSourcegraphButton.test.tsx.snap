// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewOnSourcegraphButton /> existence could not be determined  because of an authentication failure minimalUI = false renders a sign in button if showSignInButton = true 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  repoExistsOrError={[HTTPStatusError: Request to  failed with 401 Unauthorized]}
  showSignInButton={true}
  sourcegraphURL="https://test.com"
>
  <SignInButton
    className="open-on-sourcegraph test"
    rel="noopener noreferrer"
    sourcegraphURL="https://test.com"
    target="_blank"
  >
    <SourcegraphIconButton
      ariaLabel="Sign into Sourcegraph to get hover tooltips, go to definition and more"
      className="open-on-sourcegraph test"
      href="https://test.com/sign-in?close=true"
      label="Sign in to Sourcegraph"
      onClick={[Function]}
      title="Sign into Sourcegraph to get hover tooltips, go to definition and more"
    >
      <a
        aria-label="Sign into Sourcegraph to get hover tooltips, go to definition and more"
        className="open-on-sourcegraph test"
        href="https://test.com/sign-in?close=true"
        onClick={[Function]}
        title="Sign into Sourcegraph to get hover tooltips, go to definition and more"
      >
        <SourcegraphIcon>
          <svg
            viewBox="0 0 40 40"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
                fill="#F96316"
              />
              <path
                d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
                fill="#B200F8"
              />
              <path
                d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
                fill="#00B4F2"
              />
            </g>
          </svg>
        </SourcegraphIcon>
         
        Sign in to Sourcegraph
      </a>
    </SourcegraphIconButton>
  </SignInButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> existence could not be determined  because of an authentication failure minimalUI = true renders a sign in button if showSignInButton = true 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={true}
  repoExistsOrError={[HTTPStatusError: Request to  failed with 401 Unauthorized]}
  showSignInButton={true}
  sourcegraphURL="https://test.com"
>
  <SignInButton
    className="open-on-sourcegraph test"
    rel="noopener noreferrer"
    sourcegraphURL="https://test.com"
    target="_blank"
  >
    <SourcegraphIconButton
      ariaLabel="Sign into Sourcegraph to get hover tooltips, go to definition and more"
      className="open-on-sourcegraph test"
      href="https://test.com/sign-in?close=true"
      label="Sign in to Sourcegraph"
      onClick={[Function]}
      title="Sign into Sourcegraph to get hover tooltips, go to definition and more"
    >
      <a
        aria-label="Sign into Sourcegraph to get hover tooltips, go to definition and more"
        className="open-on-sourcegraph test"
        href="https://test.com/sign-in?close=true"
        onClick={[Function]}
        title="Sign into Sourcegraph to get hover tooltips, go to definition and more"
      >
        <SourcegraphIcon>
          <svg
            viewBox="0 0 40 40"
          >
            <g
              fill="none"
              fillRule="evenodd"
            >
              <path
                d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
                fill="#F96316"
              />
              <path
                d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
                fill="#B200F8"
              />
              <path
                d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
                fill="#00B4F2"
              />
            </g>
          </svg>
        </SourcegraphIcon>
         
        Sign in to Sourcegraph
      </a>
    </SourcegraphIconButton>
  </SignInButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> existence could not be determined  because of an unknown error renders a button with an error label 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  repoExistsOrError={[Error: Something unknown happened!]}
  showSignInButton={true}
  sourcegraphURL="https://test.com"
>
  <SourcegraphIconButton
    ariaLabel="Something unknown happened!"
    className="open-on-sourcegraph test"
    href="https://test.com/test@test"
    iconClassName="open-on-sourcegraph__icon--muted"
    label="Error"
    rel="noopener noreferrer"
    target="_blank"
    title="Something unknown happened!"
  >
    <a
      aria-label="Something unknown happened!"
      className="open-on-sourcegraph test"
      href="https://test.com/test@test"
      rel="noopener noreferrer"
      target="_blank"
      title="Something unknown happened!"
    >
      <SourcegraphIcon
        className="open-on-sourcegraph__icon--muted"
      >
        <svg
          className="open-on-sourcegraph__icon--muted"
          viewBox="0 0 40 40"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
              fill="#F96316"
            />
            <path
              d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
              fill="#B200F8"
            />
            <path
              d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
              fill="#00B4F2"
            />
          </g>
        </svg>
      </SourcegraphIcon>
       
      Error
    </a>
  </SourcegraphIconButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> repository does not exist on the instance renders "Configure Sourcegraph" button when pointing at sourcegraph.com 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  onConfigureSourcegraphClick={[Function]}
  repoExistsOrError={false}
  sourcegraphURL="https://sourcegraph.com"
>
  <SourcegraphIconButton
    ariaLabel="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.com"
    className="open-on-sourcegraph test"
    href="https://sourcegraph.com/test@test"
    iconClassName="open-on-sourcegraph__icon--muted"
    label="Repository not found"
    rel="noopener noreferrer"
    target="_blank"
    title="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.com"
  >
    <a
      aria-label="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.com"
      className="open-on-sourcegraph test"
      href="https://sourcegraph.com/test@test"
      rel="noopener noreferrer"
      target="_blank"
      title="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.com"
    >
      <SourcegraphIcon
        className="open-on-sourcegraph__icon--muted"
      >
        <svg
          className="open-on-sourcegraph__icon--muted"
          viewBox="0 0 40 40"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
              fill="#F96316"
            />
            <path
              d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
              fill="#B200F8"
            />
            <path
              d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
              fill="#00B4F2"
            />
          </g>
        </svg>
      </SourcegraphIcon>
       
      Repository not found
    </a>
  </SourcegraphIconButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> repository does not exist on the instance renders a "Repository not found" button when not pointing at sourcegraph.com 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  onConfigureSourcegraphClick={[Function]}
  repoExistsOrError={false}
  sourcegraphURL="https://sourcegraph.test"
>
  <SourcegraphIconButton
    ariaLabel="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.test"
    className="open-on-sourcegraph test"
    href="https://sourcegraph.test/test@test"
    iconClassName="open-on-sourcegraph__icon--muted"
    label="Repository not found"
    rel="noopener noreferrer"
    target="_blank"
    title="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.test"
  >
    <a
      aria-label="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.test"
      className="open-on-sourcegraph test"
      href="https://sourcegraph.test/test@test"
      rel="noopener noreferrer"
      target="_blank"
      title="The repository does not exist on the configured Sourcegraph instance https://sourcegraph.test"
    >
      <SourcegraphIcon
        className="open-on-sourcegraph__icon--muted"
      >
        <svg
          className="open-on-sourcegraph__icon--muted"
          viewBox="0 0 40 40"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
              fill="#F96316"
            />
            <path
              d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
              fill="#B200F8"
            />
            <path
              d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
              fill="#00B4F2"
            />
          </g>
        </svg>
      </SourcegraphIcon>
       
      Repository not found
    </a>
  </SourcegraphIconButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> repository exists on the instance renders a link to the repository on the Sourcegraph instance 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  repoExistsOrError={true}
  sourcegraphURL="https://test.com"
>
  <SourcegraphIconButton
    ariaLabel="View repository on Sourcegraph"
    className="open-on-sourcegraph test"
    href="https://test.com/test"
    rel="noopener noreferrer"
    target="_blank"
    title="View repository on Sourcegraph"
  >
    <a
      aria-label="View repository on Sourcegraph"
      className="open-on-sourcegraph test"
      href="https://test.com/test"
      rel="noopener noreferrer"
      target="_blank"
      title="View repository on Sourcegraph"
    >
      <SourcegraphIcon>
        <svg
          viewBox="0 0 40 40"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
              fill="#F96316"
            />
            <path
              d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
              fill="#B200F8"
            />
            <path
              d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
              fill="#00B4F2"
            />
          </g>
        </svg>
      </SourcegraphIcon>
       
    </a>
  </SourcegraphIconButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> repository exists on the instance renders a link with the revision when provided 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={false}
  repoExistsOrError={true}
  sourcegraphURL="https://test.com"
>
  <SourcegraphIconButton
    ariaLabel="View repository on Sourcegraph"
    className="open-on-sourcegraph test"
    href="https://test.com/test@test"
    rel="noopener noreferrer"
    target="_blank"
    title="View repository on Sourcegraph"
  >
    <a
      aria-label="View repository on Sourcegraph"
      className="open-on-sourcegraph test"
      href="https://test.com/test@test"
      rel="noopener noreferrer"
      target="_blank"
      title="View repository on Sourcegraph"
    >
      <SourcegraphIcon>
        <svg
          viewBox="0 0 40 40"
        >
          <g
            fill="none"
            fillRule="evenodd"
          >
            <path
              d="M11.5941935,5.12629921 L20.4929032,36.888189 C21.0909677,39.0226772 23.3477419,40.279685 25.5325806,39.6951181 C27.7190323,39.1105512 29.0051613,36.9064567 28.4067742,34.7722835 L19.5064516,3.00944882 C18.9080645,0.875590551 16.6516129,-0.381732283 14.4667742,0.203149606 C12.2822581,0.786771654 10.9958065,2.99149606 11.5941935,5.12598425 L11.5941935,5.12629921 Z"
              fill="#F96316"
            />
            <path
              d="M28.0722581,5.00598425 L5.7883871,29.5748031 C4.28516129,31.2314961 4.44225806,33.7647244 6.13741935,35.2327559 C7.83258065,36.7004724 10.4245161,36.5474016 11.9277419,34.8913386 L34.2116129,10.3228346 C35.7148387,8.66614173 35.5577419,6.13385827 33.8625806,4.66551181 C32.1667742,3.19653543 29.5741935,3.34992126 28.0722581,5.00566929 L28.0722581,5.00598425 Z"
              fill="#B200F8"
            />
            <path
              d="M2.82258065,18.6204724 L34.6019355,28.8866142 C36.7525806,29.5811024 39.0729032,28.4412598 39.7841935,26.3395276 C40.4970968,24.2381102 39.3293548,21.9716535 37.1777419,21.2762205 L5.39935484,11.0110236 C3.24774194,10.3159055 0.928387097,11.455748 0.216774194,13.5574803 C-0.494193548,15.6588976 0.673548387,17.9259843 2.82322581,18.6204724 L2.82258065,18.6204724 Z"
              fill="#00B4F2"
            />
          </g>
        </svg>
      </SourcegraphIcon>
       
    </a>
  </SourcegraphIconButton>
</ViewOnSourcegraphButton>
`;

exports[`<ViewOnSourcegraphButton /> repository exists on the instance renders nothing in minimal UI mode 1`] = `
<ViewOnSourcegraphButton
  className="test"
  codeHostType="test-codehost"
  getContext={[Function]}
  minimalUI={true}
  repoExistsOrError={true}
  sourcegraphURL="https://test.com"
/>
`;
